File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Deque<TreeNode> queue = new ArrayDeque<TreeNode>();
        List<List<Integer>> ans = new ArrayList<>();
        if (root == null)
            return ans; 

        queue.offer(root); 

        while (!queue.isEmpty())
        {   List<Integer> level = new ArrayList<>(); 
            int size = queue.size(); 
            for (int i = 0; i < size; i++)
                {
                    TreeNode curr = queue.poll(); 
                    if (queue!= null)
                    {
                        level.add(curr.val);
                        if (curr. left !=null)
                            queue.offer(curr.left);
                        if (curr.right!=null)
                            queue.offer(curr.right); 
                    }

                }

            if (level.size() !=0)
                ans.add(level); 
        }

     return ans;    
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-4.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        ans = []
        for index in range (len(nums)): 
            if index > 0 and nums[index] == nums[index -1]:
                continue 
            
            left = index + 1 
            right = len(nums) - 1 
            while left < right: 
                sum = nums[index] + nums[left] + nums[right]
                if sum == 0: 
                    ans.append([nums[index], nums[left], nums[right]])
                    left += 1
                    right -= 1
                    while nums[left] == nums[left - 1] and left < right: 
                        left +=1 
                    while nums[right] == nums[right + 1] and left < right: 
                        right -=1 
                elif sum < 0: 
                    left += 1
                else:
                    right -= 1

        return ans 